server/runtime/src/main/java/org/infinispan/server/security/ElytronSASLAuthenticationProvider.java

```diff
@@ -32,12 +40,25 @@
    private final String name;
    private final String serverPrincipal;
    private final Collection<String> mechanisms;
+   private final Provider[] providers;
```

Comment:
I don't think you need to keep the `providers` forever.

Reply:
I suppose the code (and `securityProviderSaslServerFactory`) could be moved to the `init()` method instead of being in the constructor.

Reply:
ok